plugins/colorbutton/plugin.js

```diff
@@ -503,22 +516,29 @@
 			 * @returns {String} Returns color in hex format, but without the hash at the beginning, e.g. `ff0000` for red.
 			 */
 			normalizeColor: function( color ) {
-				var alphaRegex = /^(rgb|hsl)a\(/g,
-					transparentRegex = /^rgba\((\s*0\s*,?){4}\)$/g,
-					isAlphaColor = alphaRegex.test( color ),
-					// Browsers tend to represent transparent color as rgba(0, 0, 0, 0), so we need to filter out this value.
-					isTransparentColor = transparentRegex.test( color ),
-					colorInstance;
-
-				// For colors with alpha channel we need to use CKEDITOR.tools.color normalization (#4351).
-				if ( isAlphaColor && !isTransparentColor ) {
-					colorInstance = new CKEDITOR.tools.color( color );
-
-					return CKEDITOR.tools.normalizeHex( colorInstance.getHex() || '' ).replace( /#/g, '' );
+				var colorInstance = ( color instanceof CKEDITOR.tools.color ) ? color : new CKEDITOR.tools.color( color, '' ),
+					isTransparent = Number( colorInstance._.alpha ) === 0;
+
+				if ( isTransparent ) {
+					return 'transparent';
 				}
 
 				// Replace 3-character hexadecimal notation with a 6-character hexadecimal notation (#1008).
-				return CKEDITOR.tools.normalizeHex( '#' + CKEDITOR.tools.convertRgbToHex( color || '' ) ).replace( /#/g, '' );
+				// It also covers other cases, like colors with alpha channel (#4351).
+				return CKEDITOR.tools.normalizeHex( colorInstance.getHex() || '' );
```

Comment:
This function is already marked as deprecated. We already have some ambiguous `normalizations` around. The one is here in `ColorBox` and another in `tools`. So I'm starting to wonder - what is the normalization process because it leads to very different results. It is related to parsing values from CSS attributes. So maybe we should separate the so-called `normalization` and extracting value from the CSS attribute?

Reply:
The issue is that removing this normalization will basically nuke tests… `CKEDITOR.tools.color` returns uppercase hexes and whole tests assume they are lowercase. So I'm not sure if it's worth it to refactor all tests.

Reply:
Or not, nothing happened :thinking: Strange, because it suggests that there's yet another normalization on the way…